cnd/src/config/settings.rs

```diff
@@ -257,4 +308,38 @@ mod tests {
                 listen: vec!["/ip4/0.0.0.0/tcp/9939".parse().unwrap()],
             })
     }
+
+    #[test]
+    fn bitcoin_defaults() {
+        let config_file = File { ..File::default() };
+
+        let settings = Settings::from_config_file_and_defaults(config_file);
+
+        assert_that(&settings)
+            .is_ok()
+            .map(|settings| &settings.bitcoin)
+            .is_equal_to(Bitcoin {
+                network: bitcoin::Network::Regtest,
+                bitcoind: Bitcoind {
+                    node_url: "http://localhost:18443".parse().unwrap(),
+                },
+            })
+    }
+
+    #[test]
+    fn ethereum_defaults() {
```

Comment:
It would be good to have tests here for the case where the user specifies the network but not the URL.

They would catch the bug you have in `derive_url_ethereum` ;)

Reply:
Please have a look at the new tests.